docs/enhancements/label-clusterimagesets.md

```diff
@@ -0,0 +1,96 @@
+---
+title: label-clusterimagesets-with-version-and-arch
+authors:
+  - "@mhrivnak"
+creation-date: 2022-02-23
+last-updated: 2022-02-23
+---
+
+# Label ClusterImageSets with version and arch
+
+
+## Summary
+
+Label each ClusterImageSet resource with two labels: the architecture of the
+release image, and the OpenShift version. This will enable assisted-service to
+understand the releases and find a particular version and architecture when
+needed.
+
+## Motivation
+
+When installing clusters with an architecture other than x86_64,
+assisted-service requires the user to specify OpenShift release images for both
+the x86_64 architecture and the target architecture, at the same version. This
+is because regardless of the target architecture, assisted-service will run the
+openshift-install binary on its own cluster as part of the process for creating
+installation manifests and ignition. For example, if a user wants to install
+OpenShift 4.9.21 on arm, they would utilize a ClusterImageSet resource that
+references `quay.io/openshift-release-dev/ocp-release:4.9.21-aarch64`.
+Assisted-service would then need to understand the version and architecture for
+that image and then locate an x86_64 image that is also version 4.9.21.
+
+### Goals
+
+* Enable assisted-service to understand the architecture and OpenShift version
+of each ClusterImageSet.
+
+### Non-Goals
+
+* Add other inspected information about a release in the form of annotations or labels.
+
+## Proposal
+
+A new controller will be added to assisted-service. It will watch
+ClusterImageSet as its primary resource. For each resource, it will ensure that
+two labels exist:
+
+`agent-install.openshift.io/openshift-version`: The output of `oc adm release info -o=jsonpath='{.metadata.version}'`
```

Comment:
In the case of having a true multi-arch image it wouldn't hurt the user to specify the architecture. It may end up being a no-op depending on the implementation in assisted-service. Right now we need the user to specify the architectures because we don't have multi-arch images.

Reply:
and heterogeneous clusters?